ec2/spark-ec2

```diff
@@ -20,6 +20,6 @@
 
 # Preserve the user's CWD so that relative paths are passed correctly to 
 #+ the underlying Python script.
-SPARK_EC2_DIR="$(dirname $0)"
+SPARK_EC2_DIR="$(dirname "$0")"
```

Comment:
`$()` invokes a subshell, so the quotes inside the subshell are processed independently of those outside.

So here we're making sure `dirname` gets exactly 1 argument by quoting the `$0`, in case the value of `$0` has spaces.

Then, we're making sure the output of `dirname` in turn is captured as 1 argument when passed to `SPARK_EC2_DIR` by quoting the whole subshell, in case that output has spaces.

Welcome to the most deceptive feature of Bash: [Word Splitting](http://mywiki.wooledge.org/WordSplitting).